lib/core/test/unit/Cardano/WalletSpec.hs

```diff
@@ -576,76 +560,6 @@ walletListTransactionsSorted wallet@(wid, _, _) _order (_mstart, _mend) history
-    , coinSelection :: CS.CoinSelection
-    } deriving Show
-
-{-------------------------------------------------------------------------------
-                        Properties of tx fee estimation
--------------------------------------------------------------------------------}
-
--- | Properties of 'estimateFeeForCoinSelection':
--- 1. There is no coin selection with a fee above the estimated maximum.
--- 2. The minimum estimated fee is no greater than the maximum estimated fee.
--- 3. Around 10% of fees are below the estimated minimum.
-prop_estimateFee :: NonEmptyList (Either String FeeGen) -> Property
-prop_estimateFee (NonEmpty results) = case actual of
-    Left err -> label "errors: all" $
-        Left err === head results
-    Right estimation@(W.FeeEstimation minFee maxFee _) ->
-        label ("errors: " <> if any isLeft results then "some" else "none") $
-        counterexample (show estimation) $
-            maxFee <= maximum (map (getRight 0) results) .&&.
-            minFee <= maxFee .&&.
-            (proportionBelow minFee results `closeTo` (1/10 :: Double))
-  where
-    actual :: Either String W.FeeEstimation
-    actual = runTest results' (W.estimateFeeForCoinSelection Nothing mockCoinSelection)
-
-    -- infinite list of CoinSelections (or errors) matching the given fee
-    -- amounts.
-    results' = fmap coinSelectionForFee <$> L.cycle results
-
-    -- Pops a pre-canned result off the state and returns it
-    mockCoinSelection
-        :: ExceptT String (State [Either String CoinSelection]) Fee
-    mockCoinSelection = fmap (Fee . feeBalance) $ ExceptT $ state (\(r:rs) -> (r,rs))
-
-    runTest vals action = evalState (runExceptT action) vals
-
-    -- Find the number of results below the "minimum" estimate.
-    countBelow minFee =
-        count ((< minFee) . getRight maxBound)
-    proportionBelow minFee xs = fromIntegral (countBelow minFee xs)
-        / fromIntegral (count isRight xs)
-    count p = length . filter p
-
-    -- get the coin amount from a Right FeeGen, or a default value otherwise.
-    getRight d = either (const d) (unCoin . unFeeGen)
-
-    -- Two fractions are close to each other if they are within 20% either way.
-    closeTo a b =
-        counterexample (show a <> " & " <> show b <> " are not close enough") $
-        property $ abs (a - b) < (1/5)
-
--- | A fee amount that has a uniform random distribution in the range 1-100.
-newtype FeeGen = FeeGen { unFeeGen :: Coin } deriving (Show, Eq)
-
-instance Arbitrary FeeGen where
-    arbitrary = FeeGen . Coin <$> choose (1,100)
-
--- | Manufacture a coin selection that would result in the given fee.
-coinSelectionForFee :: FeeGen -> CoinSelection
-coinSelectionForFee (FeeGen (Coin fee)) = mempty
-    { CS.inputs =
-        [(TxIn (Hash "") 0, TxOut (Address "") (coinToBundle (1 + fee)))]
-    , CS.outputs =
-        [TxOut (Address "") (coinToBundle 1)]
-    }
-  where
-    coinToBundle = TokenBundle.fromCoin . Coin
```

Comment:
Whoops. I was maybe a bit too fast in removing this one thinking it was something else. We likely still want something similar.